finality-aleph/src/network/component.rs

```diff
@@ -21,18 +20,8 @@ pub trait Receiver<D: Data>: Sync + Send {
 pub trait Network<D: Data>: Sync + Send {
     type S: Sender<D>;
     type R: Receiver<D>;
-    fn sender(&self) -> &Self::S;
-    fn receiver(&self) -> Arc<Mutex<Self::R>>;
-}
 
-#[async_trait::async_trait]
-impl<D: Data, CN: Network<D>> DataNetwork<D> for CN {
-    fn send(&self, data: D, recipient: Recipient) -> Result<(), SendError> {
-        self.sender().send(data, recipient)
-    }
-    async fn next(&mut self) -> Option<D> {
-        self.receiver().lock_owned().await.next().await
-    }
+    fn into(self) -> (Self::S, Self::R);
```

Comment:
Already discussed out of band, but this trait should be removed, and all its implementers should be made into various specializations of `SimpleNetwork` (perhaps now called `ComponentNetwork`). Then the latter struct should implement `Into<S, R>` and that should be enough for our purposes.